segregation/inference/comparative.py

```diff
@@ -0,0 +1,453 @@
+        / CT1_2_compl
+        * df1["compl_pop_var"].sum()
+    )
+    df2["counterfactual_compl_pop"] = (
+        df2["compl_share"].rank(pct=True).apply(df1["compl_share"].quantile)
+        / CT2_1_compl
+        * df2["compl_pop_var"].sum()
+    )
+
+    df1["counterfactual_total_pop"] = (
+        df1["counterfactual_group_pop"] + df1["counterfactual_compl_pop"]
+    )
+    df2["counterfactual_total_pop"] = (
+        df2["counterfactual_group_pop"] + df2["counterfactual_compl_pop"]
+    )
+    return df1.fillna(0), df2.fillna(0)
+
+
+def _prepare_random_label(seg_class_1, seg_class_2):
+    if hasattr(seg_class_1, "_original_data"):
+        data_1 = seg_class_1._original_data.copy()
+    else:
+        data_1 = seg_class_1.data.copy()
+    if hasattr(seg_class_2, "_original_data"):
+        data_2 = seg_class_2._original_data.copy()
+    else:
+        data_2 = seg_class_2.data.copy()
+
+    data_1["grouping_variable"] = "Group_1"
+    data_2["grouping_variable"] = "Group_2"
+
+    if isinstance(seg_class_1, SingleGroupIndex):
+
+        # This step is just to make sure the each frequecy column is integer for the approaches and from the same type in order to be able to stack them
+        data_1.loc[:, (seg_class_1.group_pop_var, seg_class_1.total_pop_var)] = (
+            data_1.loc[:, (seg_class_1.group_pop_var, seg_class_1.total_pop_var)]
+            .round(0)
+            .astype(int)
+        )
+
+        # random permutation needs the columns to have the same names
+        data_1 = data_1[
+            [seg_class_1.group_pop_var, seg_class_1.total_pop_var, "grouping_variable",]
+        ]
+        data_1.columns = ["group", "total", "grouping_variable"]
+
+        data_2.loc[:, (seg_class_2.group_pop_var, seg_class_2.total_pop_var)] = (
+            data_2.loc[:, (seg_class_2.group_pop_var, seg_class_2.total_pop_var)]
+            .round(0)
+            .astype(int)
+        )
+        data_2 = data_2[
+            [seg_class_2.group_pop_var, seg_class_2.total_pop_var, "grouping_variable",]
+        ]
+        data_2.columns = ["group", "total", "grouping_variable"]
+
+        stacked_data = pd.concat([data_1, data_2], axis=0)
+
+    elif isinstance(seg_class_1, MultiGroupIndex):
+
+        groups_list = seg_class_1.groups
+
+        for i in range(len(groups_list)):
+            data_1[groups_list[i]] = round(data_1[groups_list[i]]).astype(int)
+            data_2[groups_list[i]] = round(data_2[groups_list[i]]).astype(int)
+
+        if seg_class_1.groups != seg_class_2.groups:
+            raise ValueError("MultiGroup groups should be the same")
+
+        stacked_data = pd.concat([data_1, data_2], ignore_index=True)
+    return stacked_data
+
+
+def _estimate_random_label_difference(data):
+    stacked_data = data[0]
+    function = data[1]
+    index_args_1 = data[2]
+    index_args_2 = data[3]
+    idx_type = data[4]
+    groups = data[5]
+    approach = data[6]
+
+    if approach == 'person_permutation':
```

Comment:
Shall `black` be run on the repo to have consistent `"` vs. `'`?

Reply:
sure. I'm usually pretty good about hitting the formatter before doing a commit but not always. Could also do a black precommit hook or something i just havent gotten around to it